src/webRTCEndpoint.ts

```diff
@@ -424,17 +440,24 @@ export class WebRTCEndpoint extends (EventEmitter as new () => TypedEmitter<
         data.trackIdToMetadata = new Map<string, any>(
           Object.entries(data.trackIdToMetadata)
         );
+        data.trackIdToSimulcastConfig = new Map<string, SimulcastConfig>(
+          Object.entries(data.trackIdToSimulcastConfig)
+        );
         endpoint = this.idToEndpoint.get(data.endpointId)!;
         const oldTrackIdToMetadata = endpoint.trackIdToMetadata;
         endpoint.trackIdToMetadata = new Map([
           ...endpoint.trackIdToMetadata,
           ...data.trackIdToMetadata,
         ]);
+        endpoint.trackIdToSimulcastConfig = new Map([
+          ...Array.from(endpoint.trackIdToSimulcastConfig.entries()) as Iterable<[string, SimulcastConfig]>,
+          ...Array.from(data.trackIdToSimulcastConfig.entries()) as Iterable<[string, SimulcastConfig]>,
+        ]);
```

Comment:
This section (whole `case "tracksAdded":`) is quite complex, with a lot of copying, mutating, overwriting, and casting. We'll need to refactor this in the future. I don't want you to fix it now because you've followed the "convention." I'm just letting you know that I've added a task for this, and we'll improve it later 🙃